### PR TITLE
US038 Adding Linting Checks - update

### PR DESCRIPTION
**Pull request by Cormac-F**
- Checkstyle lint is now carried out in the build action and is available to run locally using "mvn checkstyle:check"
- Added automated testing to the build action

Screenshot 2023-12-12 at 11 43 25 am

### CODE REVIEW

**Review by CharlesM988 — APPROVED** [review]
Looks good! Glad that you got it working!

File: src/main/java/org/kainos/ea/db/DatabaseConnector.java [hunk 1]
```diff
@@ -21,22 +20,26 @@ public static Connection getConnection() throws SQLException {
...
 
-            connection = DriverManager.getConnection("jdbc:mysql://" +
-                    host + "/" + name +
+            connection = DriverManager.getConnection("jdbc:mysql://"
+                    +
+                    host
+                    + "/"
+                    + name
+                    +
                     "?useSSL=false", user, password);
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/db/DatabaseConnector.java:40]: Is the linter telling you to do this? I'd find this quite weird. I.e. + and host are on new lines, but + name are on the same line.

**Review by jkelso662 — APPROVED** [review]
Looks good! Just check my comment about a style choice your linter might be pushing you towards. Also can you update the PR title so that It includes a ticket number format that matches the other tickets? Thanks! :)

File: src/main/java/org/kainos/ea/db/DatabaseConnector.java (hunk 1 above)
Comment (Cormac-F) [reply src/main/java/org/kainos/ea/db/DatabaseConnector.java:40]: Yeah that was the linter :)

**Pull request merged**